File: elixir/day_3/test/diagnostic_test.exs
defmodule DiagnosticTest do
  use ExUnit.Case
  doctest Diagnostic

  test "power consumption" do
    input = """
    00100
    11110
    10110
    10111
    10101
    01111
    00111
    11100
    10000
    11001
    00010
    01010
    """

    numbers = Diagnostic.parse_input(input)
    assert Diagnostic.PowerConsumption.calculate(numbers) == 198
  end

  test "life support rating" do
    input = """
    00100
    11110
    10110
    10111
    10101
    01111
    00111
    11100
    10000
    11001
    00010
    01010
    """

    numbers = Diagnostic.parse_input(input)
    assert Diagnostic.LifeSupport.calculate(numbers) == 230
  end
end

File: typescript/src/day_03/__tests__/diagnostic.test.ts
import { powerConsumption,lifeSupportRating } from '../diagnostic';

describe('powerConsumption', () => {
  it('returns the power consumption of the submarine', () => {
    const inputs = [
      '00100',
      '11110',
      '10110',
      '10111',
      '10101',
      '01111',
      '00111',
      '11100',
      '10000',
      '11001',
      '00010',
      '01010'
    ]
    expect(powerConsumption(inputs)).toEqual(198)
  })
});

describe('lifeSupportRating', () => {
  it('returns the life support rating of the submarine', () => {
    const inputs = [
      '00100',
      '11110',
      '10110',
      '10111',
      '10101',
      '01111',
      '00111',
      '11100',
      '10000',
      '11001',
      '00010',
      '01010'
    ]
    expect(lifeSupportRating(inputs)).toEqual(230)
  })
});
